### PR TITLE
Addition of new fields for the payment system.

### PR DESCRIPTION
**Pull request by arek1-618**
Addition of new fields for the payment system.

### CODE REVIEW

File: install/install_schema/data.sql [hunk 1]
```diff
@@ -4730,6 +4732,11 @@ insert  into `vtiger_field`(`tabid`,`fieldid`,`columnname`,`tablename`,`generate
...
+insert  into `vtiger_field`(`tabid`,`fieldid`,`columnname`,`tablename`,`generatedtype`,`uitype`,`fieldname`,`fieldlabel`,`readonly`,`presence`,`defaultvalue`,`maximumlength`,`sequence`,`block`,`displaytype`,`typeofdata`,`quickcreate`,`quickcreatesequence`,`info_type`,`masseditable`,`helpinfo`,`summaryfield`,`fieldparams`,`header_field`,`maxlengthtext`,`maxwidthcolumn`,`visible`) values (79,2804,'ssingleordersid','vtiger_paymentsin',1,10,'ssingleordersid','FL_ORDER',1,2,'',NULL,8,251,10,'V~O',1,0,'BAS',1,'',0,'',NULL,0,0,0);
+insert  into `vtiger_field`(`tabid`,`fieldid`,`columnname`,`tablename`,`generatedtype`,`uitype`,`fieldname`,`fieldlabel`,`readonly`,`presence`,`defaultvalue`,`maximumlength`,`sequence`,`block`,`displaytype`,`typeofdata`,`quickcreate`,`quickcreatesequence`,`info_type`,`masseditable`,`helpinfo`,`summaryfield`,`fieldparams`,`header_field`,`maxlengthtext`,`maxwidthcolumn`,`visible`) values (79,2805,'payment_system','vtiger_paymentsin',1,1,'payment_system','FL_PAYMENT_SYSTEM',1,2,'',NULL,9,251,10,'V~O',1,0,'BAS',1,'',0,'',NULL,0,0,0);
```
Comment (rskrzypczak) [inline install/install_schema/data.sql]: Why don't all the fields have maximumlength?

File: api/webservice/Portal/BaseAction/ReceiveFromPaymentsSystem.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	 */
+	public function put()
+	{
+		$request = $this->controller->request;
+		$paymentSystem = $request->getByType('payment_system', 'Alnum');
+		$transactionId = $request->getByType('transaction_id', 'Alnum');
+		$paymentsInId = (new \App\Db\Query())
+			->select(['paymentsinid'])
+			->from('vtiger_paymentsin')
+			->where(['transaction_id' => $transactionId])
```
Comment (TomaszQr) [inline api/webservice/Portal/BaseAction/ReceiveFromPaymentsSystem.php]: ->where(['transaction_id' => $transactionId, 'payment_system] =>paymentSystem)

File: api/webservice/Portal/BaseAction/ReceiveFromPaymentsSystem.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	public function put()
+	{
+		$request = $this->controller->request;
+		$paymentSystem = $request->getByType('payment_system', 'Alnum');
+		$transactionId = $request->getByType('transaction_id', 'Alnum');
+		$paymentsInId = (new \App\Db\Query())
+			->select(['paymentsinid'])
+			->from('vtiger_paymentsin')
+			->where(['transaction_id' => $transactionId, 'payment_system' => $paymentSystem])
+			->scalar();
```
Comment (rskrzypczak) [inline api/webservice/Portal/BaseAction/ReceiveFromPaymentsSystem.php]: QueryGenerator 
permission to record, verify state of record

File: app/Fields/Currency.php [hunk 4]
```diff
@@ -91,6 +91,27 @@ public static function getCurrencyIdByName($currencyName)
...
+		$currencyId = null;
+		$row = (new \App\Db\Query())
+			->select(['id'])
+			->from('vtiger_currency_info')
+			->where(['currency_code' => \strtoupper($currencyCode), 'deleted' => 0])
+			->scalar();
+		if ($row) {
+			$currencyId = $row;
+		}
+		return $currencyId;
```
Comment (rskrzypczak) [inline app/Fields/Currency.php]: 1. Change name to getIdByCode()
2. Add arg $active = true
2. Replace the function's contents
    $currency = array_column(\App\Fields\Currency::getAll($active), 'id', 'currency_code');
    return $currency[\strtoupper($currencyCode)]??null;
    or add foreach

File: layouts/basic/modules/Vtiger/Edit/Field/Currency.tpl [hunk 5]
```diff
@@ -80,6 +80,24 @@
...
+			<div class="input-group">
+				{assign var="DISPLAY_FIELD_VALUE" value=$FIELD_VALUE}
+				<input name="{$FIELD_MODEL->getFieldName()}" value="{$DISPLAY_FIELD_VALUE}" type="text"
+							   class="row-fluid currencyField form-control" data-fieldinfo='{$FIELD_INFO}'
+							   data-validation-engine="validate[{if $FIELD_MODEL->isMandatory() eq true} required,{/if}funcCall[Vtiger_Base_Validator_Js.invokeValidation]]"
+							   title="{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE)}"
+							   {if !empty($SPECIAL_VALIDATOR)}data-validator={\App\Json::encode($SPECIAL_VALIDATOR)}{/if} data-decimal-separator='{$USER_MODEL->get('currency_decimal_separator')}'
+							   data-group-separator='{$USER_MODEL->get('currency_grouping_separator')}'
+							   {if $FIELD_MODEL->isEditableReadOnly()}readonly="readonly"{/if} />
+				{if $SYMBOL_PLACEMENT eq '1.0$'}
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Edit/Field/Currency.tpl]: What if $SYMBOL_PLACEMENT eq '$1.0' ?
and 
?

File: modules/Vtiger/uitypes/Currency.php [hunk 6]
```diff
@@ -96,6 +96,19 @@ public function getDetailViewDisplayValue($value, $recordId, $uiType)
...
 	}
 
+	/**
+	 * Get currency symbol by record ID.
+	 *
+	 * @param int $recordId
+	 *
+	 * @return string
+	 */
+	public function getCurrencySymbolByRecordId(int $recordId): string
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/Currency.php]: rename

**Pull request closed without merging**